src/tangent.jl

```diff
@@ -250,8 +250,34 @@ UniformBundle{N}(primal, partial::U) where {N,U} = _TangentBundle(Val{N}(), prim
 UniformBundle{N, <:Any, U}(primal, partial::U) where {N, U} = _TangentBundle(Val{N}(), primal, UniformTangent{U}(U.instance))
 UniformBundle{N, <:Any, U}(primal) where {N, U} = _TangentBundle(Val{N}(), primal, UniformTangent{U}(U.instance))
 
+
 const ZeroBundle{N, B} = UniformBundle{N, B, ZeroTangent}
 const DNEBundle{N, B} = UniformBundle{N, B, NoTangent}
+const AbstractZeroBundle{N, B} = UniformBundle{N, B, <:AbstractZero}
+
+wrapper_name(::Type{<:ZeroBundle}) = "ZeroBundle"
+wrapper_name(::Type{<:DNEBundle}) = "DNEBundle"
+wrapper_name(::Type{<:AbstractZeroBundle}) = "AbstractZeroBundle"
+
+function Base.show(io::IO, T::Type{<:AbstractZeroBundle{N, B}}) where {N,B}
+    print(io, wrapper_name(T))
+    print(io, "{$N, ")
+    show(io, B)
+    print(io, "}")
+end
+
+function Base.show(io::IO, T::Type{<:AbstractZeroBundle{N}}) where {N}
+    print(io, wrapper_name(T))
+    print(io, "{$N}")
+end
+
+function Base.show(io::IO, t::AbstractZeroBundle{N}) where N
+    print(io, wrapper_name(typeof(t)))
+    print(io, "{$N}(")
+    show(io, t.primal)
+    print(io, ")")
+end
+
 
```

Comment:
All changes in this file are from #136  and should be dropped from this PR but I am including them for now as they make debugging easier